src/main/java/pl/shop/models/CustomerDetails.java

```diff
@@ -0,0 +1,33 @@
+package pl.shop.models;
+
+import lombok.Data;
+
+import javax.persistence.*;
+import java.util.UUID;
+
+@Entity
+@Data
+public class CustomerDetails {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private UUID uuid;
+
+    @OneToOne
+    @MapsId
```

Comment:
Czy dałbyś radę wyjaśnić dlaczego użyłeś tej adnotacji ?

Reply:
@mapsId powoduje, ze id tej encji zawsze bedzie takie samo jak encji matki. https://vladmihalcea.com/the-best-way-to-map-a-onetoone-relationship-with-jpa-and-hibernate/
Tutaj wyjasnienie. Chodzi o bardzo drobna poprawe wydajnosc; bez tej adnoracji nic sie nie popsuje